rct229/rules/section5/section5rule19.py

```diff
@@ -0,0 +1,132 @@
+from rct229.data_fns.table_G3_111_fns import table_G3_1_1_1_lookup
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_area_type_window_wall_area_dict import (
+    get_area_type_window_wall_area_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.std_comparisons import std_equal
+
+MSG_WARN_MATCHED = "BUILDING IS NOT ALL NEW AND BASELINE WWR MATCHES VALUES PRESCRIBED IN TABLE G3.1.1-1. HOWEVER, THE FENESTRATION AREA PRESCRIBED IN TABLE G3.1.1-1 DOES NOT APPLY TO THE EXISTING ENVELOPE PER TABLE G3.1 BASELINE COLUMN #5 (C). FOR EXISTING ENVELOPE, THE BASELINE FENESTRATION AREA MUST EQUAL THE EXISTING FENESTRATION AREA PRIOR TO THE PROPOSED WORK. A MANUAL CHECK IS REQUIRED TO VERIFY COMPLIANCE."
+MSG_WARN_MISMATCHED = "BUILDING IS NOT ALL NEW AND BASELINE WWR DOES NOT MATCH VALUES PRESCRIBED IN TABLE G3.1.1-1. HOWEVER, THE FENESTRATION AREA PRESCRIBED IN TABLE G3.1.1-1 DOES NOT APPLY TO THE EXISTING ENVELOPE PER TABLE G3.1 BASELINE COLUMN #5(c). FOR EXISTING ENVELOPE, THE BASELINE FENESTRATION AREA MUST EQUAL THE EXISTING FENESTRATION AREA PRIOR TO THE PROPOSED WORK. A MANUAL CHECK IS REQUIRED TO VERIFY COMPLIANCE."
+WWR_THRESHOLD = 0.4
+
+
+class Section5Rule19(RuleDefinitionListIndexedBase):
+    """Rule 19 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule19, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule19.BuildingRule(),
+            index_rmr="baseline",
+            id="5-19",
+            description="For building areas not shown in Table G3.1.1-1, vertical fenestration areas for new buildings and additions shall equal that in the proposed design or 40% of gross above-grade wall area, whichever is smaller.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule19.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "$": ["building_segments"],
+                    "building_segment": [
+                        "is_all_new",
+                        "area_type_vertical_fenestration",
+                    ],
+                },
+                each_rule=Section5Rule19.BuildingRule.BuildingSegmentRule(),
+                index_rmr="baseline",
+            )
+
```

Comment:
If I am correct that the BuildingRule should be the lowest level rule, then the BuildingRule will need:
- A `get_calc_vals` method that returns 
```
{
  "all_building_segments_are_new": all([buliding_segment_b.get("is_all_new", False) for building_sement_b in find_all("building_segments[*]", building_b)]),
  "other_window_wall_areas_b: area_type_window_wall_areas_dict_b["OTHER"],
  "other_window_wall_areas_p: area_type_window_wall_areas_dict_p["OTHER"]
}
```
- A `manual_check_required` method such as:
```
def manual_check_required(self, context, calc_vals, data=None):
  return not calc_vals["all_building_segments_are_new"]
```
- A `manual_check_required_msg` method such as:
```
def manual_check_required_msg(self, context, calc_vals, data=None):
  return MSG_WARN_MATCHED 
    if std_equal(
      other_window_wall_areas_b["total_window_area"],  
      min(other_window_wall_areas_p["total_window_area"], 0.4 * other_window_wall_areas_p["total_wall_area"])
    )
    else MSG_WARN_MISMATCHED
```
- A `rule_check` method
  
```